common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/DerivedWorkData.scala

```diff
@@ -1,17 +1,33 @@
 package uk.ac.wellcome.models.work.internal
 
 case class DerivedWorkData(
-  availableOnline: Boolean
+  availableOnline: Boolean,
+  contributorAgents: List[String],
 )
 
 object DerivedWorkData {
-  def none: DerivedWorkData = DerivedWorkData(availableOnline = false)
 
-  def apply(workData: WorkData[_]): DerivedWorkData =
+  def none: DerivedWorkData =
     DerivedWorkData(
-      availableOnline = containsDigitalLocation(workData.items)
+      availableOnline = false,
+      contributorAgents = Nil
     )
 
+  def apply(data: WorkData[_]): DerivedWorkData =
+    DerivedWorkData(
+      availableOnline = containsDigitalLocation(data.items),
+      contributorAgents = contributorAgents(data.contributors)
+    )
+
+  private def contributorAgents(
+    contributors: List[Contributor[_]]): List[String] =
+    contributors.map {
+      case Contributor(_, agent, _) =>
+        val ontologyType = agent.getClass.getSimpleName
+        val label = agent.label
```

Comment:
I think it's probably worth asserting that this doesn't contain a `:` as @alexwlchan was saying yesterday

Reply:
What is the issue if there is? Some labels are likely to contain that, and if we allow it through I don't see the problem, as in practice we will be splitting at the first occurrence of `:` and checking the prefix is one of `Person`, `Agent`, `Meeting`, `Organisation`..

Reply:
Hmmm, yeah, makes sense - perhaps a comment about splitting on the first occurrence then? It wasn't obvious to me that it would work like that rather than splitting into a list which we might incorrectly assume always had 2 elements

Reply:
Ah, so my suggested assertion was that the ontology type doesn't contain a `:`, not the agent. Which admittedly seems unlikely, but is the sort of thing that could creep in – then you'd get out a different thing than you started with.

Reply:
I'm pretty sure that is an illegal character for a class name :man_shrugging: